gef.py

```diff
@@ -10368,19 +10397,31 @@ def maps(self) -> List[Section]:
         return self.__maps
 
     def __parse_maps(self) -> List[Section]:
-        """Return the mapped memory sections"""
+        """Return the mapped memory sections. If the current arch has its maps
+        method defined, then defer to that to generated maps, otherwise, try to
+        figure it out from procfs, then info sections, then monitor info
+        mem."""
+        if gef.arch.maps is not None:
+            maps = list(gef.arch.maps())
+            return maps
+
         try:
-            if is_qemu_system():
-                return list(self.__parse_info_mem())
-        except gdb.error:
-            # Target may not support this command
+            return list(self.parse_procfs_maps())
+        except:
```

Comment:
Not a fan of silent exceptions for those cases, can make debugging much harder. Replace those `pass` with something more meaningful for debug (even just logging the exception)

Reply:
Same remark for the 2 following `except:pass`

Reply:
We already had silent exceptions. This is just a method to try the 3 methods in a row. Look at `__parse_maps` in main.

```python
    def __parse_maps(self) -> List[Section]:
        """Return the mapped memory sections"""
        try:
            if is_qemu_system():
                return list(self.__parse_info_mem())
        except gdb.error:
            # Target may not support this command
            pass
        try:
            return list(self.__parse_procfs_maps())
        except FileNotFoundError:
            return list(self.__parse_gdb_info_sections())
```

If we logged, then the user would have their log filled with irrelevant crap where they might have needed the third method to get map data.

Reply:
I've made it so that it warns if none of the providers work.